src/plugins/intel_cpu/src/nodes/reference.cpp

```diff
@@ -13,8 +14,11 @@ Reference::Reference(const std::shared_ptr<ov::Node>& op, const GraphContext::CP
                                          const std::string& errorMessage) :
         Node(op, context, NgraphShapeInferFactory(op, FULL_PORT_MASK)), ovCoreNode(op), additionalErrorMessage(errorMessage) {
     if (!op->has_evaluate()) {
-        OPENVINO_THROW_NOT_IMPLEMENTED(
-            "Cannot fallback on ngraph reference implementation (Ngraph::Node::evaluate() is not implemented");
+        auto converted = util::convert_to_supported_precision(op);
+        if (!converted->has_evaluate()) {
+            OPENVINO_THROW_NOT_IMPLEMENTED(
+                "Cannot fallback on ov reference implementation (ov::Node::evaluate() is not implemented");
+        }
```

Comment:
Shouldn't we overwrite `ovCoreNode` with `converted` so that the `Reference` works with supported precisions? In that case, the CPU node will set corresponding precisions to its ports and the plugin graph logic will take care of the necessary precision conversions.